kafka/integration_test.go

```diff
@@ -0,0 +1,529 @@
+		t.Errorf("Cannot create offset for 10\n")
+	} else {
+		if offset != Offset(10) {
+			t.Errorf("Offset does not equal expected: %s != %s\n", offset, Offset(10))
+		}
+	}
+
+	// test integer offset
+	var int_offset int = 10
+	offset, err = NewOffset(int_offset)
+	if err != nil {
+		t.Errorf("Cannot create offset for 10\n")
+	} else {
+		if offset != Offset(10) {
+			t.Errorf("Offset does not equal expected: %s != %s\n", offset, Offset(10))
+		}
+	}
+
+	// test int64 offset
+	var int64_offset int64 = 10
+	offset, err = NewOffset(int64_offset)
+	if err != nil {
+		t.Errorf("Cannot create offset for 10\n")
+	} else {
+		if offset != Offset(10) {
+			t.Errorf("Offset does not equal expected: %s != %s\n", offset, Offset(10))
+		}
+	}
+
+	// test invalid string offset
+	offset, err = NewOffset("what is this offset")
+	if err == nil {
+		t.Errorf("Expected error for this string offset\n")
+	}
+
+	// test double offset
+	offset, err = NewOffset(12.15)
+	if err == nil {
+		t.Errorf("Expected error for this double offset: 12.15\n")
+	}
+
+	// test change offset via Set()
+	offset, err = NewOffset("beginning")
+	if err != nil {
+		t.Errorf("Cannot create offset for 'beginning'\n")
+	}
+	err = offset.Set("latest")
+	if err != nil {
+		t.Errorf("Cannot set offset to 'latest'\n")
+	}
+
+	// test OffsetTail()
+	tail := OffsetTail(offset)
+	t.Logf("offset tail %v\n", tail)
+
+}
+
+//Test config map APIs
+func TestConfigMapAPIs(t *testing.T) {
+	if !testconfRead() {
+		t.Skipf("Missing testconf.json")
+	}
+
+	kConfig := &ConfigMap{}
+
+	// set a good key via SetKey()
+	err := kConfig.SetKey("bootstrap.servers", testconf.Brokers)
+	if err != nil {
+		t.Errorf("Failed to set key via SetKey()")
+	}
+
+	err = kConfig.SetKey("{topic}.produce.offset.report", true)
+	if err != nil {
+		t.Errorf("Failed to set key via SetKey()")
+	}
+
+	// set a good key-value pair via Set()
+	err = kConfig.Set("group.id=test.id")
+	if err != nil {
+		t.Errorf("Failed to set key-value pair via Set()")
+	}
+
+	// negative test cases
+	// set a bad key-value pair via Set()
+	err = kConfig.Set("group.id:test.id")
+	if err == nil {
+		t.Errorf("Expected failure when setting invalid key-value pair via Set()")
+	}
+
+}
+
+//Test consumer QueryWatermarkOffsets API
+func TestConsumerQueryWatermarkOffsets(t *testing.T) {
+	if !testconfRead() {
+		t.Skipf("Missing testconf.json")
+	}
+
+	kConfig := &ConfigMap{"bootstrap.servers": testconf.Brokers,
+		"group.id": groupId}
+
+	// Create consumer
+	c, err := NewConsumer(kConfig)
+	if err != nil {
+		t.Errorf("Failed to create consumer: %s\n", err)
+		return
+	}
+	low, high, err := c.QueryWatermarkOffsets(testconf.Topic, 0, 5*1000)
+	if err != nil {
+		t.Errorf("Failed to query watermark offsets for topic %s\n", testconf.Topic)
+		return
+	} else {
+		t.Logf("Watermark offsets fo topic %s: low=%d, high=%d\n", testconf.Topic, low, high)
+	}
+}
+
+//Test producer QueryWatermarkOffsets API
+func TestProducerQueryWatermarkOffsets(t *testing.T) {
```

Comment:
Could this one and the one above be unified into a helper function that is called from each test, to avoid code duplication